src/Microsoft.DotNet.CodeFormatter.Analyzers/ProvideExplicitVariableTypeAnalyzer.cs

```diff
@@ -53,12 +53,15 @@ public override void Initialize(AnalysisContext context)
                 }
 
                 var node = (VariableDeclarationSyntax)syntaxContext.Node;
+                var model = syntaxContext.SemanticModel;
+                var token = syntaxContext.CancellationToken;
                 // C# syntax doesn't allow implicitly typed variables to have multiple declarators,
                 // but we need to handle error situations (incomplete or extra declarators).   
                 if (node.Type != null &&
                     node.Type.IsVar &&
                     !IsTypeObvious(node) &&
-                    !IsAnonymousType(node.Variables.FirstOrDefault(), syntaxContext.SemanticModel, syntaxContext.CancellationToken))
+                    !IsAnonymousType(node.Variables.FirstOrDefault(), model, token) &&
+                    !HasErrors(node, model, token))
```

Comment:
So if there's any error anywhere in the program, we wouldn't apply this rule? That sounds too strict. I'd suggest simply checking if node.Type.IsErrorType() and bailing in that case.

Also computing all the diagnostics is going to be expensive as well.